src/coprocessor/dag/batch_executor/table_scan_executor.rs

```diff
@@ -310,482 +312,296 @@ mod tests {
-            ];
-
-            let mut executor = BatchTableScanExecutor::new(
-                store.clone(),
-                Arc::new(EvalConfig::default()),
-                vec![columns_info[2].clone()],
-                key_ranges.clone(),
-                false,
-            )
-            .unwrap();
-
-            let mut result = executor.next_batch(10);
-            assert!(result.is_drained.as_ref().unwrap());
-            assert_eq!(result.data.columns_len(), 1);
-            assert_eq!(result.data.rows_len(), 5);
-            assert!(result.data[0].is_raw());
-            result.data[0].decode(&Tz::utc(), &schema[2]).unwrap();
-            assert_eq!(
-                result.data[0].decoded().as_real_slice(),
-                &[Some(5.2), None, Some(4.5), Some(0.1), Some(4.5)]
-            );
+        fn get_values_by_range(&self, start_row: usize, rows: usize) -> Vec<VectorValue> {
+            let mut pks = VectorValue::with_capacity(self.data.len(), EvalType::Int);
+            let mut foos = VectorValue::with_capacity(self.data.len(), EvalType::Int);
+            let mut bars = VectorValue::with_capacity(self.data.len(), EvalType::Real);
+            for id in start_row..start_row + rows {
+                let (handle, foo, bar) = self.data[id];
+                pks.push_int(Some(handle));
+                foos.push_int(foo);
+                bars.push_real(bar);
+            }
+            vec![pks, foos, bars]
         }
 
-        // Case 8. PK is in the middle of the schema
-        {
-            let mut executor = BatchTableScanExecutor::new(
-                store.clone(),
-                Arc::new(EvalConfig::default()),
-                vec![
-                    columns_info[1].clone(),
-                    columns_info[0].clone(),
-                    columns_info[2].clone(),
-                ],
-                key_ranges_all.clone(),
-                false,
-            )
-            .unwrap();
-
-            let mut result = executor.next_batch(10);
-            assert!(result.is_drained.as_ref().unwrap());
-            assert_eq!(result.data.columns_len(), 3);
-            assert_eq!(result.data.rows_len(), 5);
-            assert!(result.data[0].is_raw());
-            result.data[0].decode(&Tz::utc(), &schema[1]).unwrap();
-            assert_eq!(
-                result.data[0].decoded().as_int_slice(),
-                &[Some(10), Some(-5), None, None, None]
-            );
-            assert!(result.data[1].is_decoded());
-            assert_eq!(
-                result.data[1].decoded().as_int_slice(),
-                &[Some(1), Some(3), Some(4), Some(5), Some(6)]
-            );
-            assert!(result.data[2].is_raw());
-            result.data[2].decode(&Tz::utc(), &schema[2]).unwrap();
-            assert_eq!(
-                result.data[2].decoded().as_real_slice(),
-                &[Some(5.2), None, Some(4.5), Some(0.1), Some(4.5)]
-            );
+        fn expect_table_values(
```

Comment:
Please add comments for these functions that supports the test because they are actually not easy to understand and need to carefully read the code to know what it is doing.

Reply:
done.